plugin/trino-hive-hadoop2/bin/run_hive_s3_tests.sh

```diff
@@ -10,6 +10,9 @@ check_vars S3_BUCKET S3_BUCKET_ENDPOINT \
     AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
 
 cleanup_hadoop_docker_containers
+
+# Use Hadoop version 3.1 for S3 tests as the JSON SerDe class is not available in lower versions.
+export HADOOP_BASE_IMAGE="ghcr.io/trinodb/testing/hdp3.1-hive"
```

Comment:
FYI @findepi (in case you weren't already aware).

Reply:
Thanks @hashhar 

If I am reading this correctly, this removes the point of running these tests in a matrix, which we still do 
https://github.com/trinodb/trino/blob/b646676438b4368fb900434c0430e9272f88ae7b/.github/workflows/ci.yml#L207-L208


@arhimondr was it a conscious decision?
did you want to remove matrix from that CI job?

cc @nineinchnick @alexjo2144

Reply:
Yes, we should either make `testGetRecordsJson` run only on HDP3 (via surefire config probably) or remove matrix (we probably shouldn't do this to maintain coverage across distros).

Reply:
Yeah, I think we should maintain the matrix. The Hive version may matter for what exactly gets created on S3.

@arhimondr @preethiratnam 
can you please remove `export HADOOP_BASE_IMAGE=...` hack from `run_hive_s3_tests.sh`?

Reply:
Hi @findepi @hashhar I want to use the HADOOP_BASE_IMAGE 3.1 to run Hive S3 tests. This is because the JSONSerDe class is not available in the 2.6 version.

Is there a better way of configuring the JSON tests to use 3.1 version?

Reply:
@preethiratnam  We need to use test exclusions via maven profiles for that.

Reply:
Thanks, I'll raise a new PR to fix this.

Reply:
@findepi No, unfortunately it wasn't a conscious decision. I think I misread and assumed that it will only change the default value for S3 tests.

Reply:
Created https://github.com/trinodb/trino/pull/13486 to fix this.